DataLayer/BatteryFaultsData/I_BatteryFaultsData.h

```diff
@@ -38,8 +38,8 @@ class I_BatteryFaultsData : public QObject
     virtual unsigned short getErrorFlag() const = 0;
 
     /* BatteryFaults Sets */
-    void setPackageID(const unsigned char& packageID) = 0;
-    void setErrorFlag(const unsigned char& errorFlag) = 0;
+    virtual void setPackageID(const unsigned char& packageID) = 0;
```

Comment:
I realise this is a syntax error now, would the following be more appropriate @enochtsang :

`class BatteryData
: public I_BatteryData
, packageID_(7)
{
...
private:
   const unsigned char packageID_;
}`